### PR TITLE
Add Metadata DTO to Data package

### PR DESCRIPTION
**Pull request by patri-create**
### :pushpin: References
* **Issue:** [Improve networking performance](https://github.com/Sefford/Artdrian/issues/27)

### :tophat: What is the goal?

Having DTOs make your app more flexible to backend changes. Meanwhile in the backend they can fine-tuned or modify their objects and logic, that does not mean that you need to change the object in the app. You will use a representation that lives in the domain app.

Besides, I unified the files related with data layer in a Data package with the intention of isolate them in a module in the future. 

### How is it being implemented?

Add MetadataDto object and move files related to network/memory to Data package.

### CODE REVIEW

**Comment by patri-create** [comment]
I'm having issues with the tests.

1. differences with the snaps. Not sure how to create new snaps or what is happening.

image


2. Dagger missing binding. I have to take a look deeply.

image

File: app/src/test/java/com/sefford/artdrian/datasources/WallpaperRepositoryTest.kt [hunk 1]
```diff
@@ -66,7 +67,7 @@ class WallpaperRepositoryTest {
             val local = WallpaperMemoryDataSource()
             val repository = WallpaperRepository(api, local)
 
-            local.saveMetadata(listOf(SECOND_METADATA))
+            local.saveMetadata(listOf(SECOND_METADATADTO).map { it.toDomain() })
```
Comment (sdgonzalez-eb) [inline app/src/test/java/com/sefford/artdrian/datasources/WallpaperRepositoryTest.kt:70]: We don't charge for the underscore (SECOND_METADATA_DTO)

File: app/src/test/java/com/sefford/artdrian/MetadataMother.kt [hunk 2]
```diff
@@ -10,23 +11,39 @@ object MetadataMother {
...
+    val SECOND_METADATADTO = MetadataDto(
         id = SECOND_METADATA_ID,
         slug = "test_002",
         views = 100,
         downloads = 200,
         created = Date.from(LocalDate.parse("2022-11-05").atStartOfDay().toInstant(ZoneOffset.UTC)),
         updated = Date.from(LocalDate.parse("2022-11-05").atStartOfDay().toInstant(ZoneOffset.UTC)),
     )
-    val GHOST_WAVES_003 = Metadata(
+    val FIRST_METADATA = com.sefford.artdrian.model.Metadata(
```
Comment (sdgonzalez-eb) [inline app/src/test/java/com/sefford/artdrian/MetadataMother.kt:30]: Maybe you can fix this import too.

**Comment by sdgonzalez-eb** [comment]
Great job!

I see you duplicated the Metadata class for the wallpaper. 

My error here was to use a single class for all the classes. This is coupling me to the API response.

Once you arrive to the Domain layer, the first questions you should ask yourself is:

- Does this DTO makes sense on my Domain?
- What is the real information we need to extract from the DTO?
- How to make my domain models rich?

The duplication you did is a good start. Now both Metadata and MetadataDTO can evolve in different ways; however at the same time is horrible because you are commiting the same error I did with twice the work. But maybe it's the way it needs to be.

Let's start by giving it a powerful name to help you think. What if instead of "Metadata" we call them "Wallpaper"; maybe "Metadata" is something inside of the Wallpaper (or not!). I leave that exercise to you.

Also, try to make the Metadata less anemic. Try to identify what behaviors live outside of the class, and belong to it.

File: app/src/test/java/com/sefford/artdrian/datasources/WallpaperRepositoryTest.kt (hunk 1 above)
Comment (patri-create) [reply app/src/test/java/com/sefford/artdrian/datasources/WallpaperRepositoryTest.kt:70]: I'll change it

**Comment by patri-create** [comment]
The endpoint has changed. I'll open a new PR fixing that.

**Pull request closed without merging**